src/EntityMetaStorage.php

```diff
@@ -178,26 +178,20 @@ public function doPostSave(EntityInterface $entity, $update) {
     parent::doPostSave($entity, $update);
 
     $default_revision = (bool) $entity->get('emr_default_revision')->value;
-    // If the current entity being saved has been marked as the default
-    // revision, we need to go through all its other revisions and mark them
-    // as not default.
     if ($default_revision) {
-      $current = $entity->getRevisionId();
-      $revision_ids = $this->revisionIds($entity);
-      foreach ($revision_ids as $id) {
-        if ($id === $current || is_null($current)) {
-          continue;
-        }
-        /** @var \Drupal\emr\Entity\EntityMetaInterface $revision */
-        $revision = $this->loadRevision($id);
-        $revision->set('emr_default_revision', FALSE);
-        $revision->setNewRevision(FALSE);
-        $revision->markToSkipRelations();
-        $revision->setHostEntity(NULL);
-        $revision->setForcedNoRevision(TRUE);
-        $revision->save();
+      // If the saved revision was marked as default, track this in the
+      // outside table (if not already done). Unless, the relations to this
+      // revisions should be deleted, in which case it cannot stay the default.
+      if ((int) $entity->getRevisionId() !== $this->getDefaultRevisionId((int) $entity->id())) {
+        $this->trackDefaultRevision((int) $entity->id(), (int) $entity->getRevisionId());
       }
```

Comment:
This costs an extra query to check the value and one to update it, maybe we can just update it without re-checking?

Reply:
Write queries are more expensive than reads, so my comment is wrong.